data/src/history/metadata.rs

```diff
@@ -57,6 +58,54 @@ impl fmt::Display for ReadMarker {
     }
 }
 
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct MessageReferences {
+    pub timestamp: DateTime<Utc>,
+    pub id: Option<String>,
+}
+
+impl MessageReferences {
+    pub fn message_reference(
+        &self,
+        message_reference_types: &[isupport::MessageReferenceType],
+    ) -> isupport::MessageReference {
+        for message_reference_type in message_reference_types {
+            match message_reference_type {
+                isupport::MessageReferenceType::MessageId => {
+                    if let Some(id) = &self.id {
+                        return isupport::MessageReference::MessageId(id.clone());
+                    }
+                }
+                isupport::MessageReferenceType::Timestamp => {
+                    return isupport::MessageReference::Timestamp(self.timestamp);
+                }
+            }
+        }
+
+        isupport::MessageReference::None
+    }
+}
+
+impl PartialEq for MessageReferences {
+    fn eq(&self, other: &Self) -> bool {
+        self.timestamp.eq(&other.timestamp)
```

Comment:
Currently this is only used for some syntactic sugar, in `update_partial` in `data/src/history.rs` this (via `Ord`) allows `.max` to be used.  `id` isn't considered because for the purpose it's currently being used we only want to know if we have a more recent set of references.

I originally implemented it this way because `Ord` was utilized elsewhere in the refactor, but that turned out to be unnecessary and was stripped back to just the one use rather than unraveled completely.  No particular attachment to it.